aten/src/ATen/native/native_functions.yaml

```diff
@@ -3635,12 +3636,12 @@
 - func: pixel_shuffle(Tensor self, int upscale_factor) -> Tensor
   dispatch:
     CPU: pixel_shuffle_cpu
-    CompositeExplicitAutograd: math_pixel_shuffle
+    CompositeExplicitAutogradNonFunctional: math_pixel_shuffle
```

Comment:
> the point of this set of changes is to mark 'unsafe' kernels that do non-functional things

yep! The point is mostly a better error message - the `*NonFunctional` key will set things up so that this kernel does not get registered to the Lazy + XLA dispatch keys. We don't want that, because the kernel for this function decomposes into view operators.

> later that is utilized by e.g. LTC IR codegen to call into the functionalize helper

Pretty much. Well, the LTC codegen doesn't know about it - there are ~8-9 of these "problematic" decompositions in total, and I manually wrote the code to call into the functionalize helper for those ops in `ts_native_functions.cpp`. Using in codegen instead is definitely an option - I just didn't bother because the amount of boilerplate was pretty low, and I expect these kernels to change very rarely (given that there are only 8-9 "problematic decompositions" out of our 2000+ aten ops, it looks like we don't add these very often).